herbie/main.rkt

```diff
@@ -107,24 +109,28 @@
 	   (if (null? rest-splits) acc
 	       (loop (cdr rest-splits)
 		     (let ([splitpoint (car rest-splits)])
-		       `(if (< ,(list-ref (program-variables (*start-prog*)) (sp-vidx splitpoint))
+		       `(if (< ,(sp-bexpr splitpoint)
 			       ,(sp-point splitpoint))
 			    ,(program-body (alt-program (list-ref alts (sp-cidx splitpoint))))
 			    ,acc))))))
      alts splitpoints)))
 
+(define best-alt
+  (compose (curry argmin alt-cost)
+	   (curry argmins (compose errors-score alt-errors))))
+
 (define (improve-loop table fuel)
   (cond [(<= fuel 0)
 	 (debug "Ran out of fuel, reducing... " #:from 'main #:depth 2)
-	 (post-process table fuel)]
+	 (post-process table)]
```

Comment:
Even when the recursive regimes flag is on, it doesn't happen in post-process, but I guess the fuel thing was probably a relic from a time when it did.